packable/packable/src/packable/map.rs

```diff
@@ -0,0 +1,239 @@
+    /// [`Packable::UnpackError`].
+    Value(VE),
+    /// Semantic error raised when the length prefix cannot be unpacked.
+    Prefix(P),
+}
+
+impl<K, KE: fmt::Debug, VE: fmt::Debug, P: fmt::Debug> fmt::Debug for UnpackMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::DuplicateKey(_) => f.debug_tuple("DuplicateKey").finish(),
+            Self::Key(arg0) => f.debug_tuple("Key").field(arg0).finish(),
+            Self::Value(arg0) => f.debug_tuple("Value").field(arg0).finish(),
+            Self::Prefix(arg0) => f.debug_tuple("Prefix").field(arg0).finish(),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl<K, KE, VE, P> std::error::Error for UnpackMapError<K, KE, VE, P>
+where
+    KE: std::error::Error,
+    VE: std::error::Error,
+    P: std::error::Error,
+{
+}
+
+impl<K, KE, VE, P> From<Infallible> for UnpackMapError<K, KE, VE, P> {
+    fn from(err: Infallible) -> Self {
+        match err {}
+    }
+}
+
+impl<K, KE: fmt::Display, VE: fmt::Display, P: fmt::Display> fmt::Display for UnpackMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::DuplicateKey(_) => write!(f, "duplicate key in map"),
+            Self::Key(err) => write!(f, "cannot unpack key: {}", err),
+            Self::Value(err) => write!(f, "cannot unpack value: {}", err),
+            Self::Prefix(err) => write!(f, "cannot unpack prefix: {}", err),
+        }
+    }
+}
+
+/// Error type raised when a semantic error occurs while unpacking an ordered map.
+pub enum UnpackOrderedMapError<K, KE, VE, P> {
+    /// A map error.
+    Map(UnpackMapError<K, KE, VE, P>),
+    /// An unordered map.
+    Unordered,
+}
+
+impl<K, KE, VE, P> From<UnpackMapError<K, KE, VE, P>> for UnpackOrderedMapError<K, KE, VE, P> {
+    fn from(value: UnpackMapError<K, KE, VE, P>) -> Self {
+        Self::Map(value)
+    }
+}
+
+impl<K, KE: fmt::Debug, VE: fmt::Debug, P: fmt::Debug> fmt::Debug for UnpackOrderedMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Map(arg0) => f.debug_tuple("Map").field(arg0).finish(),
+            Self::Unordered => f.debug_tuple("Unordered").finish(),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl<K, KE, VE, P> std::error::Error for UnpackOrderedMapError<K, KE, VE, P>
+where
+    KE: std::error::Error,
+    VE: std::error::Error,
+    P: std::error::Error,
+{
+}
+
+impl<K, KE, VE, P> From<Infallible> for UnpackOrderedMapError<K, KE, VE, P> {
+    fn from(err: Infallible) -> Self {
+        match err {}
+    }
+}
+
+impl<K, KE: fmt::Display, VE: fmt::Display, P: fmt::Display> fmt::Display for UnpackOrderedMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Map(s) => s.fmt(f),
```

Comment:
for consistency should also be `err` I guess

Reply:
Done